jdbc/src/main/java/com/interface21/jdbc/core/JdbcTemplate.java

```diff
@@ -14,4 +19,57 @@ public class JdbcTemplate {
     public JdbcTemplate(final DataSource dataSource) {
         this.dataSource = dataSource;
     }
+
+    public void update(final String sql, final Object... args) {
+        try (Connection connection = dataSource.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            log.debug("query : {}", sql);
+            setParameters(preparedStatement, args);
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public <T> T queryForObject(final String sql, final RowMapper<T> rowMapper, final Object... args) {
+        try (Connection conn = dataSource.getConnection();
+                PreparedStatement preparedStatement = conn.prepareStatement(sql);
+        ) {
+            setParameters(preparedStatement, args);
+            log.debug("query : {}", sql);
+            try (ResultSet rs = preparedStatement.executeQuery()) {
+                if (rs.next()) {
+                    return rowMapper.mapRow(rs, 0);
+                }
+            }
+            return null;
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
```suggestion
    public <T> T queryForObject(final String sql, final RowMapper<T> rowMapper, final Object... args) {
        try (Connection conn = dataSource.getConnection();
                PreparedStatement preparedStatement = conn.prepareStatement(sql);
        ) {
            setParameters(preparedStatement, args);
            log.debug("query : {}", sql);
            try (ResultSet rs = preparedStatement.executeQuery()) {
                if (rs.next()) {
                    return rowMapper.mapRow(rs, 0);
                }

                throw new NoSuchElementException("해당 쿼리는 주어진 매개변수에 대해 아무런 결과도 반환하지 않았습니다");
            }
        } catch (SQLException e) {
            log.error(e.getMessage(), e);
            throw new RuntimeException(e);
        }
    }
```

만약 UserDao에서 Optional 처리를 추가한다면 이 부분도 바꿔줘야 할 것 같아요